### PR TITLE
[HttpKernel] Add `HttpException::fromStatusCode()`

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Issues        | -
| License       | MIT

A small helper idea that popped up when reviewing #53209

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Exception/HttpException.php [hunk 1]
```diff
@@ -29,6 +29,27 @@ public function __construct(int $statusCode, string $message = '', \Throwable $p
         parent::__construct($message, $code, $previous);
     }
 
+    public static function fromStatusCode(int $statusCode, string $message = '', \Throwable $previous = null, array $headers = [], int $code = 0): self
+    {
+        return match ($statusCode) {
+            400 => new BadRequestHttpException($message, $previous, $code, $headers),
```
Comment (alexandre-daubois) [inline src/Symfony/Component/HttpKernel/Exception/HttpException.php:35]: Shouldn't the method signature follow an arguments order similar to exception?

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpKernel/Exception/HttpException.php:35]: it follows the constructor instead, which provides local consistency (and the code is the less useful argument so it's better last to me)

**Review by fabpot — APPROVED** [review]

**Review by alexandre-daubois — APPROVED** [review]

**Comment by derrabus** [comment]
A test? 🥺

**Comment by nicolas-grekas** [comment]
This is already tested by the added code paths to me. More tests would look like testing a getter to me :)

**Review by OskarStark — APPROVED** [review]

**Comment by simondaigre** [comment]
Great idea. It will help to ignore some HTTP exceptions on Sentry. 
Could we also plug this with `#[WithHttpStatus]` attribute ?

**Review by dunglas — APPROVED** [review]
I would also add a test, with a data provider, it will catch typos etc (returning the an exception not corresponding to the passed parameter).

**Comment by nicolas-grekas** [comment]
> I would also add a test, with a data provider

alright, updated

>  Could we also plug this with #[WithHttpStatus] attribute ?

Already handled, that's what the changes on ErrorListener are about.

**Pull request merged**